dynamic/internal/fixup/properties.go

```diff
@@ -41,9 +42,35 @@ func Default(p *info.Provider) error {
 	return errors.Join(
 		fixPropertyConflict(p),
 		fixMissingIDs(p),
+		fixProviderResource(p),
 	)
 }
 
+// fixProviderResource renames any resource that would otherwise be called `Provider`,
+// since that conflicts with the package's actual `Provider` resource.
+func fixProviderResource(p *info.Provider) error {
```

Comment:
Is this only problematic if in the top-level module?

Reply:
Yes, because that's where we generate the `Provider` resource normally.